### PR TITLE
Use force merge at the end of freeze

### PR DESCRIPTION
**Pull request by icristescu**
Depends on https://github.com/mirage/index/pull/253

### CODE REVIEW

File: src/irmin-pack/pack_index.ml [hunk 1]
```diff
@@ -80,7 +82,10 @@ module Make (K : Irmin.Hash.S) = struct
 
   let v = Index.v ~cache
 
-  let add t k v = replace t k v
+  let add ?overcommit t k v =
+    match overcommit with
+    | Some true -> replace ~overcommit:() t k v
+    | _ -> replace t k v
```
Comment (pascutto) [inline src/irmin-pack/pack_index.ml:88]: Perhaps both `overcommit` options should be unified to avoid this?

File: test/irmin-pack/test_pack.ml [hunk 2]
```diff
@@ -262,7 +262,10 @@ module Pack = struct
     t.clone_index_pack ~readonly:true >>= fun (i, r) ->
     let test w =
       let adds l =
-        List.iter (fun (k, v) -> Pack.unsafe_append ~ensure_unique:true w k v) l
+        List.iter
+          (fun (k, v) ->
+            Pack.unsafe_append ~ensure_unique:true ~overcommit:false w k v)
```
Comment (pascutto) [inline test/irmin-pack/test_pack.ml:267]: If we're going with an optional argument, then maybe they're not necessary in these tests (plus the semantics should not change)

File: src/irmin-pack/pack_index.ml (hunk 1 above)
Comment (icristescu) [reply src/irmin-pack/pack_index.ml]: Done in https://github.com/mirage/irmin/pull/1183/files#diff-a8e8d436cf37ad285558a295e2e031ed0ecd868fe9490b411bbe5eefd3c9186dR85

**Comment by icristescu** [comment]
Rebased

**Pull request merged**